l2geth/internal/ethapi/api.go

```diff
@@ -798,14 +798,19 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	}
 	// Set sender address or use a default if none specified
 	var addr common.Address
```

Comment:
This will prevent users from passing `from` in the RPC request and have it be applied. The logic should just skip the `if args.From == nil` check when `rcfg.UsingOVM` is set. Setting `addr = common.HexToAddress("0x00..00")` is redundant since `var addr common.Address` initially is the zero value, which is `address(0)`

Reply:
Thank you for your thorough explanation :) Everything should be fine now.